jcommune-view/jcommune-web-view/src/test/java/org/jtalks/jcommune/test/utils/Users.groovy

```diff
@@ -118,4 +118,17 @@ abstract class Users {
         return userDao.getByUsername(username) == null
     }
 
+    def boolean isAuthenticated(HttpSession session, User user) {
+
+        //From Spring Security source code
+        def context = session.getAttribute(HttpSessionSecurityContextRepository.SPRING_SECURITY_CONTEXT_KEY)
+
+        if (context == null || ! (context instanceof SecurityContext)) {
+            return false;
+        }
+
+        def auth = (context as SecurityContext).authentication
+        return auth != null && auth.authenticated && ((auth.principal as JCUser).username.equals(user.username))
```

Comment:
Can be simplified: `return auth?.authenticated && ((auth.principal as JCUser).username == user.username)`